test/unit/math/mix/fun/trace_gen_inv_quad_form_ldlt_test.cpp

```diff
@@ -32,52 +24,125 @@ TEST(mathMixMatFun, traceGenInvQuadForm) {
   Eigen::MatrixXd c11(1, 1);
   c11 << -3;
   stan::test::expect_ad(f, c11, a11, b11);
+  stan::test::expect_ad_matvar(f, c11, a11, b11);
 
-  // tolerance very low for gradients with var; tolerance OK for fvar
-  // which uses templated prim implementation using templated prim
-  // impl for rev still has issues below gradient_grad_ = 1e0; hessian
-  // tolerance also low
+  // tolerance very low for gradients
   stan::test::ad_tolerances tols;
-  tols.gradient_grad_ = 1e3;
-  tols.hessian_hessian_ = 1e0;
-  tols.hessian_fvar_hessian_ = 1e0;
+  tols.gradient_grad_ = stan::test::relative_tolerance(0.1, 0.15);
+  tols.hessian_hessian_ = stan::test::relative_tolerance(0.1, 0.15);
+  tols.hessian_fvar_hessian_ = stan::test::relative_tolerance(0.1, 0.15);
 
   Eigen::MatrixXd a(4, 4);
-  a << 2, 3, 4, 5, 6, 10, 2, 2, 7, 2, 7, 1, 8, 2, 1, 112;
+  a << 6, 3, 4, 1, 3, 7, 2, 2, 4, 2, 7, 1, 1, 2, 1, 5;
   Eigen::MatrixXd b(4, 2);
   b << 100, 10, 0, 1, -3, -3, 5, 2;
   Eigen::MatrixXd c(2, 2);
   c.setIdentity(2, 2);
   stan::test::expect_ad(tols, f, c, a, b);
+  stan::test::expect_ad_matvar(tols, f, c, a, b);
 
   Eigen::MatrixXd d(2, 2);
   d << 1, 2, 3, 4;
   stan::test::expect_ad(tols, f, d, a, b);
+  stan::test::expect_ad_matvar(tols, f, d, a, b);
 
   Eigen::MatrixXd A(2, 2);
-  A << 2, 3, 3, 7;
+  A << 3, 1, 1, 4;
   Eigen::MatrixXd B(2, 2);
   B << 5, 6, 7, 8;
   Eigen::MatrixXd D(2, 2);
   D << 9, 10, 11, 12;
   stan::test::expect_ad(tols, f, D, A, B);
+  stan::test::expect_ad_matvar(tols, f, D, A, B);
 
   // exception tests
-  // non-square second arg
-  Eigen::MatrixXd a34(3, 4);
-  stan::test::expect_ad(f, c, a34, b);
```

Comment:
Why is this deleted?

Reply:
`a34 + a34.transpose()` cannot be computed (in the symmetrizing step) so this fails before it even hits the function.

Reply:
I could add a function back that doesn't do the symmetrize though and it should fail. This should be tested in prim anywho

Reply:
Yeah, lets add such test back in. Prim is fine.

Reply:
Alright, the problem with adding a check back is that the new `LDLT_factor` isn't doing `check_symmetric` things. It's leaving that to the functions that use it.

So there's no way to construct the LDLT factor to test `trace_gen_inv_quad_form_ldlt`, and then `LDLT_factor` is used internally only in places where we've already done the symmetry checks.

Would be 10 minutes to add a symmetry check + tests, so pretty easy. You think we should? Otherwise I'll leave it.

Reply:
Now that you merged develop in I cant even find this anymore. Im confused. It now looks like nothing was deleted so it is fine I guess.